app/src/lib/git/stash.ts

```diff
@@ -144,23 +144,68 @@ function extractBranchFromMessage(message: string): string | null {
   return match === null || match[1].length === 0 ? null : match[1]
 }
 
-/** Get the files that were changed in the given stash commit.
- *  This is different than `getChangedFiles` because stashes
- *  have _3 parents(!!!)_
+/**
+ * The SHA for the [empty tree](https://stackoverflow.com/questions/9765453/is-gits-semi-secret-empty-tree-object-reliable-and-why-is-there-not-a-symbolic).
+ *
+ */
+const NullTreeSHA = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+
+/**
+ * Get the files that were changed in the given stash commit.
  *
- * TODO: support showing untracked files in the diff
+ * Runs `git diff` twice, once for tracked changes in the stash
+ * and once for untracked changes in the stash.
+ * This is different than `getChangedFiles` because stashes
+ * have _3 parents(!!!)_
  */
 export async function getStashedFiles(
   repository: Repository,
-  sha: string
+  committish: string
 ): Promise<ReadonlyArray<CommittedFileChange>> {
   // opt-in for rename detection (-M) and copies detection (-C)
   // this is equivalent to the user configuring 'diff.renames' to 'copies'
   // NOTE: order here matters - doing -M before -C means copies aren't detected
+  const baseArgs = ['diff', '-C', '-M', '--name-status', '-z']
+  const trackedArgs = [...baseArgs, committish, `${committish}^`, '--']
```

Comment:
I've got a repro here that feels confusing, and might be related to this area:

 - remove a tracked file from the repository, see the diff in the working directory

<img width="1290" src="https://user-images.githubusercontent.com/359239/56240733-925ed880-606a-11e9-824c-a0e2081c4778.png">

 - switch branches, stash in previous branch, then return to previous branch

 - view current stash, see deleted file has correct diff but is considered "added"

<img width="1032" src="https://user-images.githubusercontent.com/359239/56240734-925ed880-606a-11e9-8acf-96b8ee9151c5.png">

I had a look at the dev tools and confirmed this was coming from this Git operation:

<img width="623" src="https://user-images.githubusercontent.com/359239/56240962-0d27f380-606b-11e9-945f-087801b21918.png">

I think the order of these should be switched around:

```
$ git diff -C -M --name-status stash@{0} stash@{0}^
A       tslint.json
git diff -C -M --name-status stash@{0}^ stash@{0}
D       tslint.json
```

Reply:
fixed in 84bbde0